inst/plumber/v1/endpoints.R

```diff
@@ -1,601 +1,613 @@
+#* @param release_version:[chr] date when the data was published in YYYYMMDD format
+#* @param ppp_version:[chr] ppp year to be used
+#* @param version:[chr] Data version. Defaults to most recent version. See api/v1/versions
+#* @serializer json
+function(req) {
+  params <- req$argsQuery
+  params$lkup <- lkups$versions_paths[[req$argsQuery$version]]
+  params$version <- NULL
+  if (params$country == "all") {
+    promises::future_promise({
+      do.call(pipapi::ui_cp_charts, params)
+    }, seed = TRUE)
+  } else {
+    do.call(pipapi::ui_cp_charts, params)
+  }
+}
+
+#* Return Country Profile - Downloads
+#* @get /api/v1/cp-download
+#* @param country:[chr] Country ISO3 code
+#* @param povline:[dbl] Poverty Line
+#* @param release_version:[chr] date when the data was published in YYYYMMDD format
+#* @param ppp_version:[chr] ppp year to be used
+#* @param version:[chr] Data version. Defaults to most recent version. See api/v1/versions
+#* @serializer switch
+function(req) {
+  params <- req$argsQuery
+  params$lkup <- lkups$versions_paths[[req$argsQuery$version]]
+  params$version <- NULL
+  params$format  <- NULL
+
+  if (params$country == "all") {
+    out <- promises::future_promise({
+      tmp <- do.call(pipapi::ui_cp_download, params)
+      attr(tmp, "serialize_format") <- req$argsQuery$format
+      tmp
+    }, seed = TRUE)
+  } else {
+    out <- do.call(pipapi::ui_cp_download, params)
+    attr(out, "serialize_format") <- req$argsQuery$format
+  }
+  out
+}
+
+# UI Endpoints: Survey metadata  ------------------------------------------
+
+#* Return data for the Data Sources page
+#* @get /api/v1/survey-metadata
+#* @param country:[chr] Country ISO3 code
+#* @param release_version:[chr] date when the data was published in YYYYMMDD format
+#* @param ppp_version:[chr] ppp year to be used
+#* @param version:[chr] Data version. Defaults to most recent version. See api/v1/versions
+#* @serializer json list(na="null")
+function(req) {
+  params <- req$argsQuery
+  params$lkup <- lkups$versions_paths[[req$argsQuery$version]]
+  params$version <- NULL
+  do.call(pipapi::ui_svy_meta, params)
+}
+
+#* Return valid years
+#* @get /api/v1/valid-years
+#* @param release_version:[chr] date when the data was published in YYYYMMDD format
+#* @param ppp_version:[chr] ppp year to be used
+#* @param version:[chr] Data version. Defaults to most recent version. See api/v1/versions
+#* @serializer switch
+function(req) {
+  params <- req$argsQuery
+  params$lkup <- lkups$versions_paths[[params$version]]
+  pipapi::valid_years(data_dir = params$lkup$data_root)
+}
+
+
+#* Return citation
+#* @get /api/v1/citation
+#* @param version:[chr] Data version. Defaults to most recent version. See api/v1/versions
+#* @param release_version:[chr] date when the data was published in YYYYMMDD format
+#* @param ppp_version:[chr] ppp year to be used
+#* @serializer switch
+function(req) {
+  params <- req$argsQuery
+  pipapi::citation_from_version(params$version)
+}

```

Comment:
I simplified this since `return_correct_version` is already called at the filter level, and `version` is an element of `req$argsQuery`